### PR TITLE
Progress Bars

### PR DESCRIPTION
**Pull request by KrishnaTech6**
fixes in progress bar for Post Answer
-fixed redundant code
-used same sealed Result class for Loading

Added search progress bar

### CODE REVIEW

**Comment by KrishnaTech6** [comment]
https://github.com/doubtless46/doubtless-app/assets/96565622/d4e5739c-538c-4f88-b528-0717e1d981a8

Search fragment progress bar

**Comment by KrishnaTech6** [comment]
https://github.com/doubtless46/doubtless-app/assets/96565622/bc2d6460-a760-400a-80f9-976665f52024

post Answer progress bar @sidsharma2002

**Comment by KrishnaTech6** [comment]
https://github.com/doubtless46/doubtless-app/assets/96565622/89e5f6e2-2e67-4a77-947b-a8c955641772

added progress bar in viewdoubtsfragment @sidsharma2002

File: app/src/main/java/com/doubtless/doubtless/screens/search/SearchFragment.kt [hunk 1]
```diff
@@ -85,7 +85,13 @@ class SearchFragment : Fragment() {
 
             searchJob = CoroutineScope(Dispatchers.Main).launch {
 
-                if (it.toString().length <= 4) return@launch
+                //if (it.toString().length <= 4) return@launch
+                //I feel its useless because there were problems with smaller keywords,
```
Comment (sidsharma2002) [inline app/src/main/java/com/doubtless/doubtless/screens/search/SearchFragment.kt:89]: you can remove the whole comment.

Comment (KrishnaTech6) [reply app/src/main/java/com/doubtless/doubtless/screens/search/SearchFragment.kt:89]: bhaiya jaise 3 letter waaala keyword krna ho to like - dsa

File: app/src/main/java/com/doubtless/doubtless/screens/answers/AnswersViewModel.kt [hunk 2]
```diff
@@ -41,9 +41,12 @@ class AnswersViewModel(
 
     fun publishAnswer(publishAnswerRequest: PublishAnswerRequest) =
         viewModelScope.launch(Dispatchers.Main) {
+
+            PublishAnswerUseCase.Result.Loading(true)
```
Comment (sidsharma2002) [inline app/src/main/java/com/doubtless/doubtless/screens/answers/AnswersViewModel.kt:45]: _publishAnswerStatus.postValue(PublishAnswerUseCase.Result.Loading)

**Comment by KrishnaTech6** [comment]
kindly check @sidsharma2002

**Pull request merged**